tests/rkt_net_test.go

```diff
@@ -838,3 +846,123 @@ func NewTestNetLongName() testutils.Test {
 		testNetCustomNatConnectivity(t, nt)
 	})
 }
+
+/*
+ * mockFlannelNetwork creates fake flannel network status file and configuration pointing to this network.
+ * We won't have connectivity, but we could check if: netName was correct and if default gateway was set.
+ */
+func mockFlannelNetwork(t *testing.T, ctx *testutils.RktRunCtx) (string, networkTemplateT, error) {
+	// write fake flannel info
+	subnetPath := filepath.Join(ctx.DataDir(), "subnet.env")
+	file, err := os.Create(subnetPath)
+	if err != nil {
+		return "", networkTemplateT{}, err
+	}
+	mockedFlannel := strings.Join([]string{
+		"FLANNEL_NETWORK=11.11.0.0/16",
+		"FLANNEL_SUBNET=11.11.3.1/24",
+		"FLANNEL_MTU=1472",
+		"FLANNEL_IPMASQ=true",
+	}, "\n")
+	if _, err = file.WriteString(mockedFlannel); err != nil {
+		return "", networkTemplateT{}, err
+	}
+
+	file.Close()
+
+	// write net config for "flannel" based network
+	ntFlannel := networkTemplateT{
+		Name:       "rkt.kubernetes.io",
+		Type:       "flannel",
+		SubnetFile: subnetPath,
+		Delegate: &delegateTemplateT{
+			IsDefaultGateway: true,
+		},
+	}
+
+	netdir := prepareTestNet(t, ctx, ntFlannel)
+
+	return netdir, ntFlannel, nil
+}
+
+/*
+ * NewNetPreserveNetNameTest checks if netName is set if network is configured via flannel
+ */
+func NewNetPreserveNetNameTest() testutils.Test {
+	return testutils.TestFunc(func(t *testing.T) {
+		ctx := testutils.NewRktRunCtx()
+		defer ctx.Cleanup()
+
+		netdir, ntFlannel, err := mockFlannelNetwork(t, ctx)
+		if err != nil {
+			t.Errorf("Can't mock flannel network: %v", err)
+		}
+
+		defer os.RemoveAll(netdir)
+		defer os.Remove(ntFlannel.SubnetFile)
+
+		startList, resumeContainer := make(chan bool), make(chan bool)
```

Comment:
to make it clear that the values aren't actually used please use `struct{}` instead of `bool`